### ISSUE TITLE
Openstack driver doesn't work with 0.16.0 version of libcloud when provisioning instances

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt#20090]
When running salt-cloud to provision instances with apache-libcloud version `0.16.0`, the following error is returned:
```
[DEBUG   ] Sending event - data = {'_stamp': '2015-01-26T23:50:09.625118', 'event': 'requesting instance', 'kwargs': {'profile': 'rackspace-test', 'image': u'Ubuntu 14.04 LTS (Trusty Tahr) (PVHVM)', 'name': 'nt-testing', 'size': u'2 GB Performance'}}
Error: There was a profile error: Error creating nt-testing on OpenStack

The following exception was thrown by libcloud when trying to run the initial deployment: 400 Bad Request Invalid config_drive provided.
```
Salt-cloud versions report:
```
# salt-cloud --versions
            Salt: 2015.2.0-267-g7792b57
          Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
          Jinja2: 2.7.2
        M2Crypto: 0.21.1
  msgpack-python: 0.3.0
    msgpack-pure: Not Installed
        pycrypto: 2.6.1
         libnacl: Not Installed
          PyYAML: 3.10
           ioflo: Not Installed
           PyZMQ: 14.0.1
            RAET: Not Installed
             ZMQ: 4.0.4
            Mako: Not Installed
 Apache Libcloud: 0.16.0
```
When I removed the newer version of libcloud and replaced it with 0.15.1, provisioning works just fine and as expected.

Versions Report for working master:
```
# salt-cloud --versions
           Salt: 2015.2.0-267-g7792b57
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.3.0
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.0.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: Not Installed
Apache Libcloud: 0.15.1
```
Note: `--list-images` and `--list-sizes` works just fine with the newer version of libcloud.

ping @techhat and @s0undt3ch

**Comment by rallytime** [comment]
Looks like this is the case for `apache-libcloud==0.17.0` as well. I am bumping the severity here.

Deleting images and `--list-images` is working, though, just not provisioning.

**Comment by rallytime** [comment]
I just tested this again and it looks to be be fixed at least as of 2015.5.1, if not sooner. I did not change any cloud configs and after updating to `apache-libcloud==0.17.0` I was able to create a VM just fine. I suspect this error was related to #19923 and fixed by #22280, though I don't know for sure. 

Versions Report:
```
# salt-cloud --versions
            Salt: 2015.5.1-48-g7024789
          Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
          Jinja2: 2.7.3
        M2Crypto: 0.21.1
  msgpack-python: 0.4.4
    msgpack-pure: Not Installed
        pycrypto: 2.6.1
         libnacl: Not Installed
          PyYAML: 3.11
           ioflo: Not Installed
           PyZMQ: 14.0.1
            RAET: Not Installed
             ZMQ: 4.0.4
            Mako: 0.9.1
 Apache Libcloud: 0.17.0
```
